ipywidgets/static/widgets/js/manager.js

```diff
@@ -280,11 +280,26 @@ define([
         /**
          * Handle when a comm is opened.
          */
-        return this.create_model({
-            model_name: msg.content.data.model_name,
-            model_module: msg.content.data.model_module,
-            comm: comm,
-        }).catch(utils.reject("Couldn't create a model.", true));
+        // For < 4.x
+        var model;
+        if (msg.content.data.model_name) {
```

Comment:
It is means that we would need to support different versions of the back-end in the frontend of ipywidgets. I don't think it is the way to go in the long run... It brings again the question of packaging of nbextensions that depend on kernel code: have one version of the js for each kernel.